airflow/serialization/helpers.py

```diff
@@ -38,7 +40,21 @@ def is_jsonable(x):
         else:
             return True
 
+    max_length = conf.getint("core", "max_templated_field_length")
+
     if not is_jsonable(template_field):
+        if len(str(template_field)) > max_length:
+            rendered = redact(str(template_field), name)
+            return (
+                "Truncated. You can change this behaviour in [core]max_templated_field_length. "
+                f"{rendered[:max_length - 79]!r}... "
+            )
         return str(template_field)
     else:
+        if template_field and len(str(template_field)) > max_length:
+            rendered = redact(str(template_field), name)
```

Comment:
Do we want to use `json.dumps` here?

Also we should reduce the stringify calls (either `str` or `json.dumps`) to exactly once for each time this function is called. The call can be expensive.

Reply:
Using json.dumps brought up some extra quotes. Reduced str usage